kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java

```diff
@@ -102,14 +103,16 @@ public Response getProcessInstanceImage(@javax.ws.rs.core.Context HttpHeaders he
             @ApiParam(value = "identifier of the process instance that image should be loaded for", required = true, example = "123") @PathParam(PROCESS_INST_ID) Long procInstId,
             @ApiParam(value = "svg completed node color", required = false, example = COMPLETED_COLOR) @QueryParam(SVG_NODE_COMPLETED_COLOR) @DefaultValue(COMPLETED_COLOR) String svgNodeCompletedColor,
             @ApiParam(value = "svg completed node border color", required = false, example = COMPLETED_BORDER_COLOR) @QueryParam(SVG_NODE_COMPLETED_BORDER_COLOR) @DefaultValue(COMPLETED_BORDER_COLOR) String svgNodeCompletedBorderColor,
-            @ApiParam(value = "svg active node border color", required = false, example = ACTIVE_BORDER_COLOR) @QueryParam(SVG_NODE_ACTIVE_COLOR) @DefaultValue(ACTIVE_BORDER_COLOR) String svgActiveNodeBorderColor) {
+            @ApiParam(value = "svg active node border color", required = false, example = ACTIVE_BORDER_COLOR) @QueryParam(SVG_NODE_ACTIVE_COLOR) @DefaultValue(ACTIVE_BORDER_COLOR) String svgActiveNodeBorderColor,
+                                            @ApiParam(value = "show or hidden instance badges", required = false, example = SHOW_INSTANCE_BADGES_DEFAULT) @QueryParam(SVG_INSTANCE_BADGES_SHOW) @DefaultValue(SHOW_INSTANCE_BADGES_DEFAULT) String showBadges) {
         Variant v = getVariant(headers);
         Header conversationIdHeader = buildConversationIdHeader(containerId, context, headers);
         try {
+            Boolean badgesFlag = SHOW_INSTANCE_BADGES_DEFAULT.equals(showBadges) ? false : Boolean.parseBoolean(showBadges);
```

Comment:
This basically means `if "false".equals(showBadges) then false else Boolean.parseBoolean(showBadges);`. So in case showBadges is "false", we return false, in case it's "true", we return boolean with value true. So it can be just `Boolean.parseBoolean(showBadges)`. No need for branching here.

Reply:
ok

Reply:
@bxf12315 Actually one thing I just realized is that you can make showBadges of type `boolean` and then the conversion happens automatically. See:
https://github.com/kiegroup/droolsjbpm-integration/blob/8bd68e30df985ede1d935a51bb95683ad4a3d1c0/kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/ProcessResource.java#L934